src/main/tables.ts

```diff
@@ -170,41 +157,41 @@ export async function importCSV(
   aidColumns: string[],
   signal: AbortSignal,
 ): Promise<{ aborted: boolean }> {
-  const client = new pg.Client(connectionConfig);
-  await client.connect();
-
-  const columnsSQL = columns.map((column) => `"${column.name}" ${column.type}`).join(', ');
-  console.info(`Table schema: ${columnsSQL}.`);
-
-  const fieldProcessors = columns.map((column) => getFieldProcessor(column.type, parseOptions));
-
   try {
-    await client.query(`BEGIN`);
-    // TODO: should we worry about (accidental) SQL-injection here?
-    await client.query(`DROP TABLE IF EXISTS "${tableName}"`);
-    await client.query(`CREATE TABLE "${tableName}" (${columnsSQL})`);
-    await copyRowsFromCsvFile(client, signal, fileName, fieldProcessors, tableName);
-    if (aidColumns.length > 0) {
-      if (aidColumns.includes(ROW_INDEX_COLUMN)) {
-        await client.query(`ALTER TABLE "${tableName}" ADD COLUMN IF NOT EXISTS "${ROW_INDEX_COLUMN}" SERIAL`);
+    await sql.begin(async (sql) => {
+      await sql`DROP TABLE IF EXISTS ${sql(tableName)}`;
+
+      const columnsSQL: SqlFragment[] = columns.map((column, i) =>
+        i > 0
+          ? sql`, ${sql(column.name)} ${sql.unsafe(column.type)}`
+          : sql`${sql(column.name)} ${sql.unsafe(column.type)}`,
+      );
+      await sql`CREATE TABLE ${sql(tableName)} (${columnsSQL})`;
+
+      const fieldProcessors = columns.map((column) => getFieldProcessor(column.type, parseOptions));
+      await copyRowsFromCsvFile(sql, signal, fileName, fieldProcessors, tableName);
+
+      if (aidColumns.length > 0) {
+        if (aidColumns.includes(ROW_INDEX_COLUMN)) {
+          await sql`ALTER TABLE ${sql(tableName)} ADD COLUMN IF NOT EXISTS ${sql(ROW_INDEX_COLUMN)} SERIAL`;
+        }
+
+        const aidColumnsSQL: SqlFragment[] = aidColumns.map((aidColumn, i) =>
+          i > 0 ? sql`, '${sql(aidColumn)}'` : sql`'${sql(aidColumn)}'`,
+        );
+
+        await sql`CALL diffix.mark_personal('${sql(tableName)}', ${aidColumnsSQL})`;
+      } else {
+        await sql`CALL diffix.mark_public('${sql(tableName)}')`;
       }
-      const aidColumnsSQL = aidColumns.map((aidColumn) => `'"${aidColumn}"'`).join(', ');
-      await client.query(`CALL diffix.mark_personal('"${tableName}"', ${aidColumnsSQL});`);
-    } else {
-      await client.query(`CALL diffix.mark_public('"${tableName}"');`);
-    }
-    await client.query(`GRANT SELECT ON "${tableName}" TO "${postgresConfig.trustedUser}"`);
```

Comment:
is removing grant select intended here? it's not necessary because we give them SUPERUSER, but I think it's safer to not rely on that and grant.

Reply:
Accident. Thanks for the catch.